### ISSUE TITLE
I get unsatisfying results when reproducing your results!

### ISSUE THREAD

**Issue by hellohawaii** [xinntao/BasicSR#83]
I got unsatisfying results when trying to reproduce your results!

The following is what I did:
I used the matlab downsample kernel, and I only used the DIV2K as the training dataset . I used the GAN-loss and the RRDB_PSNR_x4.pth pretrained model. In fact, I just clone this repo and modify the path in the train_ESRGAN.json!
The following is my train_ESRGAN.json:
file:///home/liushuzhi/Desktop/fuxian/BasicSR-master/codes/options/train/train_ESRGAN.json

The following is what I got:
The visual quality is less satisfying compared with yours! 
Your result is:
file:///home/liushuzhi/Desktop/fuxian/ESRGAN-master/results/Author_result/baboon_bicLRx4_rlt.png
My result is:
file:///home/liushuzhi/Desktop/fuxian/ESRGAN-master/results/My_result/baboon_bicLRx4_rlt.png
The ground truth is:
file:///home/liushuzhi/Desktop/fuxian/superresolution_dataset/Set14/Set14_raw_image/baboon.png
As for the PSNR and SSIM, I got PSNR_Y: PSNR: 24.976743 dB; SSIM: 0.673742 when I tested on Set14, but when I tested your model I got PSNR: 24.495635 dB; SSIM: 0.654736.

In the issues you mentioned that the downsampling filter implemented by python is different from the one implemented by matlab. Do you think this may cause my problem? I think your method should work for both two filters! You mentioned that you used more data in your paper while I just used DIV2K. Is this very important? If I used more data, it would take too much time to train!

**Comment by xinntao** [comment]
For me, the results seem OK if you use only DIV2K dataset. 
I think the extra gap is from the datasets. Using more data will not increase the time compared with your current settings. Because the training is counted by iterations instead of epochs.

I do not think the filter is the key factor.

**Comment by Feiyu-Zhang** [comment]
> I got unsatisfying results when trying to reproduce your results!
> 
> The following is what I did:
> I used the matlab downsample kernel, and I only used the DIV2K as the training dataset . I used the GAN-loss and the RRDB_PSNR_x4.pth pretrained model. In fact, I just clone this repo and modify the path in the train_ESRGAN.json!
> The following is my train_ESRGAN.json:
> {
> "name": "001_RRDB_ESRGAN_x4_DIV2K" // please remove "debug_" during training
> , "use_tb_logger": true
> , "model":"srragan"
> , "scale": 4
> , "gpu_ids": [0]
> 
> , "datasets": {
> "train": {
> "name": "DIV2K"
> , "mode": "LRHR"
> , "dataroot_HR": "/home/liushuzhi/Desktop/fuxian/superresolution_dataset/DIV2K/DIV2K_train_HR_sub_image"
> , "dataroot_LR": "/home/liushuzhi/Desktop/fuxian/superresolution_dataset/DIV2K/DIV2K_train_LR_sub_image"
> , "subset_file": null
> , "use_shuffle": true
> , "n_workers": 8
> , "batch_size": 16
> , "HR_size": 128
> , "use_flip": true
> , "use_rot": true
> }
> , "val": {
> "name": "val_set14_part"
> , "mode": "LRHR"
> , "dataroot_HR": "/home/liushuzhi/Desktop/fuxian/superresolution_dataset/Set14/Set14_raw_image"
> , "dataroot_LR": "/home/liushuzhi/Desktop/fuxian/superresolution_dataset/Set14/Set14_LR_image"
> }
> }
> 
> , "path": {
> "root": "/home/liushuzhi/Desktop/fuxian/BasicSR-master"
> // , "resume_state": "../experiments/debug_002_RRDB_ESRGAN_x4_DIV2K/training_state/16.state"
> , "pretrain_model_G": "../experiments/pretrained_models/RRDB_PSNR_x4.pth"
> }
> 
> , "network_G": {
> "which_model_G": "RRDB_net" // RRDB_net | sr_resnet
> , "norm_type": null
> , "mode": "CNA"
> , "nf": 64
> , "nb": 23
> , "in_nc": 3
> , "out_nc": 3
> , "gc": 32
> , "group": 1
> }
> , "network_D": {
> "which_model_D": "discriminator_vgg_128"
> , "norm_type": "batch"
> , "act_type": "leakyrelu"
> , "mode": "CNA"
> , "nf": 64
> , "in_nc": 3
> }
> 
> , "train": {
> "lr_G": 1e-4
> , "weight_decay_G": 0
> , "beta1_G": 0.9
> , "lr_D": 1e-4
> , "weight_decay_D": 0
> , "beta1_D": 0.9
> , "lr_scheme": "MultiStepLR"
> , "lr_steps": [50000, 100000, 200000, 300000]
> , "lr_gamma": 0.5
> 
> ```
> , "pixel_criterion": "l1"
> , "pixel_weight": 1e-2
> , "feature_criterion": "l1"
> , "feature_weight": 1
> , "gan_type": "vanilla"
> , "gan_weight": 5e-3
> 
> //for wgan-gp
> // , "D_update_ratio": 1
> // , "D_init_iters": 0
> // , "gp_weigth": 10
> 
> , "manual_seed": 0
> , "niter": 5e5
> , "val_freq": 5e3
> ```
> 
> }
> 
> , "logger": {
> "print_freq": 200
> , "save_checkpoint_freq": 5e3
> }
> }
> 
> The following is what I got:
> The visual quality is less satisfying compared with yours!
> Your result is:
> ![baboon_bicLRx4_rlt](https://user-images.githubusercontent.com/37056530/54065752-21a0f100-4260-11e9-93f2-459c3f5f50b4.png)
> My result is:
> ![baboon_bicLRx4_rlt](https://user-images.githubusercontent.com/37056530/54065746-18178900-4260-11e9-9c0c-eb3092b3d0ba.png)
> The ground truth is:
> ![baboon](https://user-images.githubusercontent.com/37056530/54065760-2d8cb300-4260-11e9-97cc-846f2621f66c.png)
> As for the PSNR and SSIM, I got PSNR_Y: PSNR: 24.976743 dB; SSIM: 0.673742 when I tested on Set14, but when I tested your model I got PSNR: 24.495635 dB; SSIM: 0.654736.
> 
> In the issues you mentioned that the downsampling filter implemented by python is different from the one implemented by matlab. Do you think this may cause my problem? I think your method should work for both two filters! You mentioned that you used more data in your paper while I just used DIV2K. Is this very important? If I used more data, it would take too much time to train!

Hi~How many epochs will roughly Set14(e.g. Monkey image) occur? I approximately trained 5000 epoch. I got the image as follow. Are you the same?
![baboon_bicLRx4_265000](https://user-images.githubusercontent.com/34130100/57502312-c487f100-731d-11e9-921f-3c0c1588daa3.png)

**Comment by xinntao** [comment]
@Feiyu-Zhang 
1) 5k is not enough, at least, > 300k.
2) Your figure indicates there must be something wrong, which is not from inadequate training.

**Comment by Feiyu-Zhang** [comment]
> @Feiyu-Zhang
> 
> 1. 5k is not enough, at least, > 300k.
> 2. Your figure indicates there must be something wrong, which is not from inadequate training.

1. Oh~ There could be a misunderstanding. I mean 5K epochs. 
2. I will check my code.
Thanks!

**Comment by Slayerxxx** [comment]
> > @Feiyu-Zhang
> > 
> > 1. 5k is not enough, at least, > 300k.
> > 2. Your figure indicates there must be something wrong, which is not from inadequate training.
> 
> 1. Oh~ There could be a misunderstanding. I mean 5K epochs.
> 2. I will check the code.
>    Thanks!

I guess @xinntao means 300k iterations at least to produce satisfactory results
